### PR TITLE
Apply REST API compatibility testing for the :modules

### PR DESCRIPTION
**Pull request by jakelandis**

### CODE REVIEW

**Comment by jakelandis** [comment]
@elasticmachine update branch

File: modules/analysis-common/build.gradle [hunk 1]
```diff
@@ -23,3 +24,12 @@ restResources {
 dependencies {
   compileOnly project(':modules:lang-painless')
 }
+
+tasks.named("yamlRestCompatTest").configure {
```
Comment (mark-vieira) [inline modules/analysis-common/build.gradle:28]: One thing I realized recently is while the transform tasks are version-specific, this isn't. So when we bump the major we'll need to remove this if appropriate.

**Review by mark-vieira — COMMENTED** [review]
Boy, I'm surprised how little this involved. No need for transforms for any of these modules? Sweet 👍

**Review by pgomulka — APPROVED** [review]
LGTM 
@mark-vieira we will add transformations together with `the compatible change` to make the tests pass

File: modules/analysis-common/build.gradle (hunk 1 above)
Comment (jakelandis) [reply modules/analysis-common/build.gradle:28]: I am thinking that maybe we want to make this source set and task name version specific (i.e. include v7 somewhere in there), like the transform task. I can follow up in a different PR. 

Hopefully there will very few black listed tests once all compatibility is applied, but your point still stands.

**Pull request merged**